projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid-navigation.service.ts

```diff
@@ -226,16 +242,39 @@ export class IgxHierarchicalGridNavigationService extends IgxGridNavigationServi
         }
     }
 
-    public performTab(currentRowEl, rowIndex, visibleColumnIndex) {
-        if (!this.grid.rowList.find(row => row.index === rowIndex + 1) && this.grid.parent &&
-        this.grid.unpinnedColumns[this.grid.unpinnedColumns.length - 1].visibleIndex === visibleColumnIndex) {
+    public performTab(currentRowEl, rowIndex, visibleColumnIndex, isSummaryRow = false) {
+        const summaryRows = this.grid.summariesRowList.toArray();
+        const hasSummaries = summaryRows.length > 0;
+        const isLastDataRow = rowIndex === this.grid.verticalScrollContainer.igxForOf.length - 1;
+        const nextIsDataRow = this.grid.dataRowList.find(row => row.index === rowIndex + 1) ;
+        const isLastColumn =  this.grid.unpinnedColumns[this.grid.unpinnedColumns.length - 1].visibleIndex === visibleColumnIndex;
+        const isLastSummaryRow = hasSummaries && isSummaryRow;
+        if (!nextIsDataRow && !(isLastDataRow && hasSummaries) && isLastColumn && !isSummaryRow) {
+            // navigating in child, next is not summary
             this.navigateDown(currentRowEl, rowIndex, 0);
+        } else if (isLastSummaryRow && isLastColumn && this.grid.parent) {
+            // navigating in child summary, next is parent summary or next parent row
+            const parent = this.grid.parent;
```

Comment:
If there is a next sibling child grid it should move to the first cell inside it, otherwise here would focus the parent summary cell.